### ISSUE TITLE
`SubscriberBuilder::with_ansi` is incorrectly gated by the `ansi` feature in tracing-subscriber 0.3.18

### ISSUE THREAD

**Issue by eric-seppanen** [tokio-rs/tracing#2996]
### Version

```
tracing v0.1.40
tracing-subscriber v0.3.18
```

### Platform
```
Linux 6.2.0-39-generic #40~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Thu Nov 16 10:53:04 UTC 2 x86_64 x86_64 x86_64 GNU/Linux

```

### Crates

`tracing-subscriber`

### Description

The `tracing-subscriber` 0.3.18 docs claim that `SubscriberBuilder::with_ansi()` can be used even when the `ansi` feature is disabled. And indeed this is true on master. But in the released 0.3.18 code this is not true: commit 049ad730c did not remove the `#[cfg(feature = "ansi")]` from the method.

The 0.3.18 version of the code is here: https://github.com/tokio-rs/tracing/blob/8b7a1dde69797b33ecfa20da71e72eb5e61f0b25/tracing-subscriber/src/fmt/mod.rs#L623-L637

**Comment by eric-seppanen** [comment]
The commit on master is slightly different: 1cb523b87

So perhaps this was an error while merging a cherry-pick?

**Comment by eric-seppanen** [comment]
Fixed by #3020.
